ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -1718,6 +1722,34 @@ public Table apply(org.apache.hadoop.hive.metastore.api.Table table) {
     }
   }
 
+  /**
+   * Validate if given materialized view has SELECT privileges for current user
+   * @param cachedMVTable
+   * @return false if user does not have privilege otherwise true
+   * @throws HiveException
+   */
+  private boolean checkPrivillegeForMV(final Table cachedMVTable) throws HiveException{
+    List<String> colNames =
+        cachedMVTable.getAllCols().stream()
+            .map(FieldSchema::getName)
+            .collect(Collectors.toList());
+
+    HivePrivilegeObject privObject = new HivePrivilegeObject(cachedMVTable.getDbName(),
+        cachedMVTable.getTableName(), colNames);
+    List<HivePrivilegeObject> privObjects = new ArrayList<HivePrivilegeObject>();
+    privObjects.add(privObject);
+    try {
+      SessionState.get().getAuthorizerV2().
+          checkPrivileges(HiveOperationType.QUERY, privObjects, privObjects, new HiveAuthzContext.Builder().build());
```

Comment:
Can we check the privileges for all MVs used by the query at once so we do not need multiple round trips?